src/Button/DigitizeButton/DigitizeButton.jsx

```diff
@@ -552,12 +552,20 @@ class DigitizeButton extends React.Component {
       drawType,
       map,
       onDrawEnd,
-      onDrawStart
+      onDrawStart,
+      digitizeLayerName
     } = this.props;
 
     let geometryFunction;
     let type = drawType;
+    let digitizeLayer = MapUtil.getLayerByName(map, digitizeLayerName);
 
+    // this checks to see if the digitizeLayer  is in the map 
+    //and if not adds it from the state to the map
+    if (!digitizeLayer) {
+      map.addLayer(this.state.digitizeLayer);
```

Comment:
I don't think we have state in react-geo, @dnlkoch

Reply:
but the component already has a state. I will see @dnlkoch though to get his opinion.

Reply:
Everything is fine here, it's the already existing component state, not a global application state.